qpc/cred/test_cred_add.py

```diff
@@ -36,70 +40,48 @@ def setup_class(cls):
         subparser = argument_parser.add_subparsers(dest="subcommand")
         cls.command = CredAddCommand(subparser)
 
-    def setup_method(self, _test_method):
-        """Create test setup."""
-        write_server_config(DEFAULT_CONFIG)
-        # Temporarily disable stderr for these tests, CLI errors clutter up
-        # nosetests command.
-        self.orig_stderr = sys.stderr
-        sys.stderr = HushUpStderr()
-        if os.path.isfile(TMP_KEY):
-            os.remove(TMP_KEY)
-        with open(TMP_KEY, "w", encoding="utf-8") as test_sshkey:
-            test_sshkey.write("fake ssh keyfile.")
-
-    def teardown_method(self, _test_method):
-        """Remove test setup."""
-        # Restore stderr
-        sys.stderr = self.orig_stderr
-        if os.path.isfile(TMP_KEY):
-            os.remove(TMP_KEY)
-
     def test_add_req_args_err(self):
         """Testing the add credential command required flags."""
-        with pytest.raises(SystemExit):
-            sys.argv = ["/bin/qpc", "credential", "add", "--name", "credential1"]
+        args = ["/bin/qpc", "credential", "add", "--name", "credential1"]
+        with pytest.raises(SystemExit), patch.object(sys, "argv", args):
             CLI().main()
 
     def test_add_no_type(self):
         """Testing the add credential without type flag."""
-        with pytest.raises(SystemExit):
-            sys.argv = [
-                "/bin/qpc",
-                "credential",
-                "add",
-                "--name",
-                "credential1",
-                "--username",
-                "foo",
-                "--password",
-            ]
+        args = [
+            "/bin/qpc",
+            "credential",
+            "add",
+            "--name",
+            "credential1",
+            "--username",
+            "foo",
+            "--password",
+        ]
+        with pytest.raises(SystemExit), patch.object(sys, "argv", args):
```

Comment:
❤️  all these improvements around `sys.argv`

Reply:
If I understood correctly, this is a better approach because it isolates the modification of sys.argv to this specific test through the context manager. So after the thest the sys.argv values are automatically restored. Is that accurate ?

Reply:
Yes, that is correct @nicolearagao! 😃